features/netsocket/nsapi_dns.cpp

```diff
@@ -40,6 +102,16 @@ nsapi_addr_t dns_servers[DNS_SERVERS_SIZE] = {
                   0,0, 0,0, 0x1c,0x04, 0xb1,0x2f}},
 };
 
+static DNS_CACHE *dns_cache[MBED_CONF_NSAPI_DNS_CACHE_SIZE];
+static uint16_t dns_message_id = 1;
+static int dns_unique_id = 1;
+static DNS_QUERY *dns_query_queue[DNS_QUERY_QUEUE_SIZE];
+// Protects cache shared between blocking and asynchronous calls
+static rtos::Mutex dns_cache_mutex;
+// Protects from several threads running asynchronous DNS
+static rtos::Mutex dns_mutex;
```

Comment:
Should these be PlatformMutex:es instead to support non-RTOS compilation?

Reply:
Yes, they can be. Changed them into PlatformMutex:es.